rest_framework/fields.py

```diff
@@ -576,10 +577,20 @@ def run_validation(self, data=empty):
         return super(CharField, self).run_validation(data)
 
     def to_internal_value(self, data):
-        return six.text_type(data)
+        value = six.text_type(data)
+
+        if self.trim_whitespace:
+            return value.strip()
+
+        return value
 
     def to_representation(self, value):
-        return six.text_type(value)
+        representation = six.text_type(value)
+
+        if self.trim_whitespace:
+            return representation.strip()
```

Comment:
@tomchristie thought about that but saw that EmailField's to_representation does `.strip()` as well.